nibabel/streamlines/trk.py

```diff
@@ -0,0 +1,821 @@
+
+        # Check for errors
+        if nb_scalars_per_point != int(nb_scalars_per_point):
+            msg = "Nb. of scalars differs from one point to another!"
+            raise DataError(msg)
+
+        if nb_properties_per_streamline != int(nb_properties_per_streamline):
+            msg = "Nb. of properties differs from one streamline to another!"
+            raise DataError(msg)
+
+        self.header[Field.NB_STREAMLINES] = self.nb_streamlines
+        self.header[Field.NB_SCALARS_PER_POINT] = nb_scalars_per_point
+        self.header[Field.NB_PROPERTIES_PER_STREAMLINE] = nb_properties_per_streamline
+
+        # Overwrite header with updated one.
+        self.file.seek(self.beginning, os.SEEK_SET)
+        self.file.write(self.header.tostring())
+
+
+class TrkFile(TractogramFile):
+    """ Convenience class to encapsulate TRK file format.
+
+    Note
+    ----
+    TrackVis (so its file format: TRK) considers the streamline coordinate
+    (0,0,0) to be in the corner of the voxel whereas NiBabel's streamlines
+    internal representation (Voxel space) assume (0,0,0) to be in the
+    center of the voxel.
+
+    Thus, streamlines are shifted of half a voxel on load and are shifted
+    back on save.
+    """
+
+    # Constants
+    MAGIC_NUMBER = b"TRACK"
+    HEADER_SIZE = 1000
+    READ_BUFFER_SIZE = 87382 * 4  # About 4 Mb when no scalars nor properties.
+
+    def __init__(self, tractogram, header=None):
+        """
+        Parameters
+        ----------
+        tractogram : :class:`Tractogram` object
+            Tractogram that will be contained in this :class:`TrkFile`.
+
+        header : dict, optional
+            Metadata associated to this tractogram file.
+
+        Notes
+        -----
+        Streamlines of the tractogram are assumed to be in *RAS+*
+        and *mm* space where coordinate (0,0,0) refers to the center
+        of the voxel.
+        """
+        if header is None:
+            header_rec = TrkWriter.create_empty_header()
+            header = dict(zip(header_rec.dtype.names, header_rec[0]))
+
+        super(TrkFile, self).__init__(tractogram, header)
+
+    @classmethod
+    def get_magic_number(cls):
+        """ Return TRK's magic number. """
+        return cls.MAGIC_NUMBER
+
+    @classmethod
+    def support_data_per_point(cls):
+        """ Tells if this format supports saving data per point. """
+        return True
+
+    @classmethod
+    def support_data_per_streamline(cls):
+        """ Tells if this format supports saving data per streamline. """
+        return True
+
+    @classmethod
+    def is_correct_format(cls, fileobj):
+        """ Check if the file is in TRK format.
+
+        Parameters
+        ----------
+        fileobj : string or file-like object
+            If string, a filename; otherwise an open file-like object
+            pointing to TRK file (and ready to read from the beginning
+            of the TRK header data).
```

Comment:
I don't know what happened but this was addressed: https://github.com/nipy/nibabel/pull/391/files#diff-5fcb7f3f6ff12c6bbadc9636bd6bfd8dR508